pallets/funding/src/tests/2_evaluation.rs

```diff
@@ -0,0 +1,552 @@
+			let evaluations = default_evaluations();
+
+			inst.create_auctioning_project(project_metadata, issuer, evaluations);
+		}
+
+		#[test]
+		fn multiple_evaluating_projects() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let project1 = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+			let project2 = default_project_metadata(inst.get_new_nonce(), ISSUER_2);
+			let project3 = default_project_metadata(inst.get_new_nonce(), ISSUER_3);
+			let project4 = default_project_metadata(inst.get_new_nonce(), ISSUER_4);
+			let evaluations = default_evaluations();
+
+			inst.create_auctioning_project(project1, ISSUER_1, evaluations.clone());
+			inst.create_auctioning_project(project2, ISSUER_2, evaluations.clone());
+			inst.create_auctioning_project(project3, ISSUER_3, evaluations.clone());
+			inst.create_auctioning_project(project4, ISSUER_4, evaluations);
+		}
+
+		#[test]
+		fn plmc_price_change_doesnt_affect_evaluation_end() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let project_metadata = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+
+			// Decreasing the price before the end doesn't make a project over the threshold fail.
+			let target_funding =
+				project_metadata.minimum_price.saturating_mul_int(project_metadata.total_allocation_size);
+			let target_evaluation_usd = Percent::from_percent(10) * target_funding;
+
+			let evaluations = vec![(EVALUATOR_1, target_evaluation_usd).into()];
+			let evaluation_plmc = MockInstantiator::calculate_evaluation_plmc_spent(evaluations.clone());
+			let evaluation_existential = evaluation_plmc.accounts().existential_deposits();
+			inst.mint_plmc_to(evaluation_plmc);
+			inst.mint_plmc_to(evaluation_existential);
+
+			let project_id = inst.create_evaluating_project(project_metadata.clone(), ISSUER_1);
+			inst.evaluate_for_users(project_id, evaluations.clone()).unwrap();
+
+			let old_price = <TestRuntime as Config>::PriceProvider::get_price(PLMC_FOREIGN_ID).unwrap();
+			PRICE_MAP.with_borrow_mut(|map| map.insert(PLMC_FOREIGN_ID, old_price / 2.into()));
+
+			inst.start_auction(project_id, ISSUER_1).unwrap();
+
+			// Increasing the price before the end doesn't make a project under the threshold succeed.
+			let evaluations = vec![(EVALUATOR_1, target_evaluation_usd / 2).into()];
+			let evaluation_plmc = MockInstantiator::calculate_evaluation_plmc_spent(evaluations.clone());
+			let evaluation_existential = evaluation_plmc.accounts().existential_deposits();
+			inst.mint_plmc_to(evaluation_plmc);
+			inst.mint_plmc_to(evaluation_existential);
+
+			let project_id = inst.create_evaluating_project(project_metadata.clone(), ISSUER_2);
+			inst.evaluate_for_users(project_id, evaluations.clone()).unwrap();
+
+			let old_price = <TestRuntime as Config>::PriceProvider::get_price(PLMC_FOREIGN_ID).unwrap();
+			PRICE_MAP.with_borrow_mut(|map| map.insert(PLMC_FOREIGN_ID, old_price * 2.into()));
```

Comment:
Is it possible that the old price was already the initial price / 2 so now initial_price / 2 * 2 = initial_price?

Reply:
Can we add a "set_price" function to the PriceProvider that does set the price in PRICE_MAP. Just for a cleaner approach.

Reply:
We're creating a second project, evaluating on that, and THEN storing the old price, so we don't care about the initial_price no?

Reply:
Its more that the second `let old_price` is the price that is set on the previous PRICE_MAP insert, so I was wondering if this was intended behaviour. So its currently the case that for example:
PLMC price at line 53 is 2
Then at line 56 its set to 1
Then PLMC price at line 68 is still 1.
And PLMC price is then set to 2 again at 71.

Reply:
Yea thats fine. 
What we care is :
Case 1) we evaluate at price x, round ends with price x/2
Case 2) we evaluate at price x, round ends with price x*2.

Remember that the evaluation is made in USD terms, which don't change when we change the price of PLMC, and we are minting the required PLMC to evaluate after each price change.

Makes sense?